tokio/src/runtime/io/scheduled_io.rs

```diff
@@ -534,9 +538,12 @@ impl Future for Readiness<'_> {
                     // still didn't return `Poll::Ready`.
                     let tick = TICK.unpack(curr) as u8;
 
+                    // Safety: We don't need to acquire the lock here because
+                    // the `waiter.interest` never changes.
+                    let interest = unsafe { (*waiter.get()).interest };
```

Comment:
Could it be shared in this scenario? You removed the comment that says it can't - is that because it actually can or just because you don't need it?